packages/stylelint-plugin/src/rules/lib/tokensInFunctions.ts

```diff
@@ -0,0 +1,202 @@
+import postcssValueParser from "postcss-value-parser"
+import { deprecatedTokenReplacements } from "../../../deprecatedTokens"
+import {
+  cantFindReplacementTokenForDeprecatedMessage,
+  noMatchingRgbParamsVariableMessage,
+} from "../../messages"
+import { CurrentKaizenToken, KaizenToken } from "../../types"
+import { transformDecl } from "../../util/functionTransformer"
+import { kaizenTokensByName } from "../../util/kaizenTokens"
+import {
+  parseVariable,
+  replaceTokenInVariable,
+  stringifyVariable,
+} from "../../util/variableUtils"
+
+/*
+  We want to use percentages consistently, so, given a string that may be a percentage or a decimal, always return the decimal form.
+  This differs from the function below, `convertPercentage`, in that this expects a valid ratio value, e.g. a decimal number OR a percentage.
+  `convertPercentage` only supports percentage values, with an optional percentage sign, meaning it can't disambiguate between decimals and percentages.
+*/
+const parsePercentageOrDecimal = (
+  percentageOrDecimal: string
+): number | "NaN" => {
+  if (percentageOrDecimal.indexOf("%") !== -1) {
+    return parsePercentage(percentageOrDecimal)
+  } else {
+    const parsed = parseFloat(percentageOrDecimal)
+    if (isNaN(parsed)) {
+      return "NaN"
+    } else {
+      return parsed
+    }
+  }
+}
+
+/**
+ * Given a string like "50%", "40%", "30", "15" etc (you don't have to pass in the percentage sign), return a decimal like 0.5, 0.4, 0.3 etc
+ */
+const parsePercentage = (value: string): number | "NaN" => {
+  // Add-alpha can be used without a percentage sign, that won't fly.
+  // Fix the percentage sign, then convert to a decimal for usage in RGBA
+  const parsed = parseInt(value.replace(/\%/, ""), 10)
```

Comment:
Should we do parseFloat?

```suggestion
  const parsed = parseFloat(value.replace(/\%/, ""))
```